Realm/RLMRealm.h

```diff
@@ -91,6 +93,26 @@ NS_ASSUME_NONNULL_BEGIN
 + (instancetype)realmWithURL:(NSURL *)fileURL;
 
 /**
+ Open a Realm asynchronously on a background queue, and then invoke a callback
+ on success or failure.
+
+ Opening a Realm asynchronously may be useful if, for example, a time-consuming
+ migration might be necessary. If `configuration.syncConfiguration` is set, the
+ callback block will only be called after the Realm has successfully downloaded
+ its remote content available at the time the operation began.
+
+ @param configuration A configuration object to use when opening the Realm.
+ @param callbackQueue The dispatch queue on which the callback should be run.
+ @param callback      A callback block. If the Realm was successfully opened,
+                      it will be passed in as an argument.
+                      Otherwise, an `NSError` describing what went wrong will be
+                      passed to the block instead.
+ */
++ (void)openAsynchronouslyWithConfiguration:(RLMRealmConfiguration *)configuration
```

Comment:
Neither of those are entirely accurate either, since the _initial_ open doesn't happen on the provided queue. I'm thinking abbreviating to `asyncOpenWithConfiguration:callbackQueue:callback:`. There's not much value in having the unabridged word, especially if the Swift API uses `asyncOpen` too.